src/Mvc/Mvc.Razor/src/MvcRazorLoggerExtensions.cs

```diff
@@ -42,7 +42,7 @@ static MvcRazorLoggerExtensions()
                 LogLevel.Debug,
                 new EventId(2, "ViewCompilerEndCodeGeneration"),
                 "Code generation for the Razor file at '{FilePath}' completed in {ElapsedMilliseconds}ms.",
-                skipEnabledCheck: true);
+                new LogDefineOptions() { SkipEnabledCheck = true });
```

Comment:
Can we just make a single `LogDefineOptions` instance that sets `SkipEnabledCheck = true` instead of newing a new one up everywhere?

Reply:
Didn't see a good place to set this up that can be used throughout all the different projects/namespaces, I'm open to suggestions. Alternatively, we can have a class level private option to avoid some repeated instantiation.

Reply:
The `class`-level option sounds good to me since there's no real reason to create more than one `LogDefineOptions` instance per file that logs. Could share an `internal` instance throughout an assembly or put one `public` instance fairly low in our dependency graph but I'm not sure we need to go that far. Do you agree @eerhardt❔

Reply:
I wouldn’t go as far as making something public. I just saw a lot of duplicated creations that could be cached. I don’t think we need to get to a single one. But sharing at some level would be good.

Reply:
Updated accordingly: https://github.com/dotnet/aspnetcore/pull/34033/commits/fd25b6b74979960b53672a10bc8da2f1c479e352